conversion.c

```diff
@@ -0,0 +1,122 @@
+// ------------------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// Ayisha Sameera,GET.
+// ------------------------------------------------------------------
+// conversion.c
+// Functions to convert a decimal number to binary and hexadecimal.
+// ------------------------------------------------------------------------------------------------
+#pragma warning (disable:4996)
+#include <stdbool.h>
+#include <malloc.h>
+#include <limits.h>
+#include "conversion.h"
+#include <math.h>
+#include <string.h>
+#include <stdlib.h>
+
+// Returns the nearest bit length.
+int NearestBitLen (int digit) {
+   if (digit <= 8) return 8;               // 1 byte
+   else if (digit <= 16) return 16;        // 2 byte
+   else return 32;                         // 4 bytes, maximum value 
+}
+
+// Converts the given input to binary.
+char* Binary (char* input) {

```

Comment:
You are using string manipulation to convert to Binary, which (according to the assignment specification) is incorrect. Integer is a more suited data type for both your input and output.

Reply:
This is valid for both Binary and Hex